consensus/src/round_manager_test.rs

```diff
@@ -1022,3 +1022,48 @@ fn safety_rules_crash() {
         node.next_proposal().await;
     });
 }
+
+#[test]
+fn echo_timeout() {
+    let mut runtime = consensus_runtime();
+    let mut playground = NetworkPlayground::new(runtime.handle().clone());
+    let mut nodes = NodeSetup::create_nodes(&mut playground, runtime.handle().clone(), 4);
+    runtime.spawn(playground.start());
+    timed_block_on(&mut runtime, async {
+        // clear the message queue
+        for node in &mut nodes {
+            node.next_proposal().await;
+        }
+        // timeout 3 nodes
+        for node in &mut nodes[1..] {
+            node.round_manager
+                .process_local_timeout(1)
+                .await
+                .unwrap_err();
+        }
+        let node_0 = &mut nodes[0];
+        // node 1 doesn't timeout and should echo the timeout after 2 timeout message
```

Comment:
Why 2 timeout messages? Node 0 should wait for 3 timeout messages for 2f + 1 timeout certficate?

Reply:
this is to test the echo timeout behavior added in https://github.com/aptos-labs/aptos-core/pull/2187

Reply:
you mean node 0, not node 1?